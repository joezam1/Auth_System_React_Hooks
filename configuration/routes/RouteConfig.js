const routeConfiguration = Object.freeze({
    homePath:'/',
    authLoginPath:'/auth/login',
    authRegisterPath:'/auth/register',
    authLogoutPath:'/auth/logout',
    privateCustomerDashboardPath:'/private/customer/dashboard',
    privateCustomerOrdersPath:'/private/customer/orders',
    privateCustomerSilverOffersPath:'/private/customer/silverOffers',
    privateCustomerGoldOffersPath:'/private/customer/goldOffers',
    privateCustomerPremiumOffersPath:'/private/customer/premiumOffers',
    privateCustomerPremiumClubMemberPath:'/private/customer/premiumClubMember',
    privateCustomerNewsletterPath:'/private/customer/newsletter',
    privateCustomerManagerOrdersPath:'/private/customer/ManagerOrders',
    privateCustomerContactSuppliersPath:'/private/customer/contact/suppliers',
    storeOrdersPath:'/store/orders',
    storeInvoicesPath:'/store/invoices',
    storeWarehousePath:'/store/warehouse',
    storeAdvertisingPath:'/store/advertising'

});

export default routeConfiguration;