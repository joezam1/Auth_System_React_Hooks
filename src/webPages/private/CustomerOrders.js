import React from 'react';
import { Link } from 'react-router-dom';
import RouteConfig from '../../../configuration/routes/RouteConfig.js';


export default function CustomerOrders(){
    return(<div className='customer-orders-section'>

        <span>This is the customer Orders Section.</span>
        <br/>
        <h3>Go To</h3>
               <span><Link to={RouteConfig.home} data-testid="link-customer-order-home-id"> Home </Link></span>
               <br/>
               <span><Link to={RouteConfig.privateCustomerDashboard} data-testid="link-customer-order-dashboard-id">Customer Dashboard</Link></span>
               <br />
    </div>)
}