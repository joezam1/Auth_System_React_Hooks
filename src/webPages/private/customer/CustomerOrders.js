import React, { useEffect, useState } from 'react';
import RouteConfig from '../../../../configuration/routes/RouteConfig.js';
import LinkButtonPrivateRedirect from '../../../components/LinkButtonPrivateRedirect.js';
import ComponentAuthorizationService from '../../../services/privateWebPagesMediator/ComponentAuthorizationService.js';
import RolePermission from '../../../library/stringLiterals/RolePermission.js';
import UserRole from '../../../library/enumerations/UserRole.js';
import Helpers from '../../../library/common/Helpers.js';
import InputCommonInspector from '../../../services/validators/InputCommonInspector.js';


//Test: DONE
export default function CustomerOrders() {
    const [username, setUsername]= useState('');
    const [customerType, setCustomerType] = useState('');
    const [authorization, setAuthorization] = useState('hide');
    const [canRead, setStateRead] = useState('hide');
    const [canWrite, setStateWrite]= useState('hide');
    const [canCreate, setStateCreate] = useState('hide');
    const [canEdit, setStateEdit] = useState('hide')
    const [canDelete, setStateDelete] = useState('hide');
    const [canReadWrite, setReadWrite]= useState('hide');
    const [canWriteCreate, setWriteCreate]= useState('hide');
    const [canReadWriteCreate, setReadWriteCreate]= useState('hide');

    useEffect(()=>{
        let userInfo = ComponentAuthorizationService.getUserInfo();
        console.log('userInfo:', userInfo);
        if(InputCommonInspector.inputExist(userInfo)){
            setUsername(userInfo.username);
            setCustomerType( UserRole[userInfo.roles[0]] );
        }

        let roleIsAuthorized = ComponentAuthorizationService.roleIsAuthorized(CustomerOrders.name);
        console.log('LinkButtonPrivateRedirect-useEffect-roleIsAuthorized', roleIsAuthorized);
        if(roleIsAuthorized){ setAuthorization('block') ; }
        let allPermissions = ComponentAuthorizationService.getAllApprovedPermissions(CustomerOrders.name)
        console.log('CustomerOrders-allPermissions', allPermissions);
        if(InputCommonInspector.inputExist(allPermissions)){
            resolveApprovedPermissions(allPermissions);
        }

    }, []);

    function resolveApprovedPermissions(permissionsArray){
        let readWriteLatest = 'hide';
        let readWriteCreateLatest = 'hide';
        let writeCreateLatest = 'hide';
        for( let a = 0; a < permissionsArray.length; a++ ){
            if(permissionsArray[a] === RolePermission.READ){ setStateRead(permissionsArray[a]); }
            if(permissionsArray[a] === RolePermission.WRITE){ setStateWrite(permissionsArray[a]); }
            if(permissionsArray[a] === RolePermission.CREATE){ setStateCreate(permissionsArray[a]); }
            if(permissionsArray[a] === RolePermission.EDIT){ setStateEdit(permissionsArray[a]); }
            if(permissionsArray[a] === RolePermission.DELETE){ setStateDelete(permissionsArray[a]); }

            if(permissionsArray[a] === RolePermission.READ || permissionsArray[a] === RolePermission.WRITE){
                readWriteLatest = permissionsArray[a];
            }
            if(permissionsArray[a] === RolePermission.READ || permissionsArray[a] === RolePermission.WRITE || permissionsArray[a] === RolePermission.CREATE){
                readWriteCreateLatest = permissionsArray[a];
            }
            if( permissionsArray[a] === RolePermission.WRITE || permissionsArray[a] === RolePermission.CREATE){
                writeCreateLatest = permissionsArray[a];
            }
        }
        setReadWrite(readWriteLatest);
        setReadWriteCreate(readWriteCreateLatest);
        setWriteCreate(writeCreateLatest);
    }


    return (<div className={'customer-orders-section webpage ' + authorization}>
    <div className='outerLayout'>
        <div className='header-container'>
            <div className='header-title'>
                <span>  Hi <strong> {username}</strong></span>
            </div>
            <div className='header-title'>
                <span> Status: <strong> {Helpers.formatStringFirstLetterCapital(customerType) } </strong> </span>
            </div>
        </div>
        <div className={'container ' + canRead}>
            <h2>Customer Orders Section.</h2>
            <div className='top-navigation-bar'>
                <h3>Go To</h3>
                <LinkButtonPrivateRedirect redirectToLocation={RouteConfig.privateCustomerDashboardPath} buttonText=" Customer Dashboard " />
            </div>
        </div>


        <div className={'container ' + canWriteCreate }>
            <button type="button" id="button3" name="add" className= { "btnCreate "+ canCreate}> Add New Order </button>

            <button type="button" id="button5" name="edit" className={ "btnWrite " + canWrite}> Write in Calendar</button>
        </div>

        <div className="table-container">
            <div className='table'>
                <div className='tableHeading'>
                    <div className='row'>
                        <div className='tableHead'>
                            Selector
                        </div>
                        <div className='tableHead'>
                            Date Order
                        </div>
                        <div className='tableHead'>
                            Order Number
                        </div>
                        <div className='tableHead'>
                            Total Amount
                        </div>
                        <div className='tableHead'>
                            Status
                        </div>
                        <div className='tableHead'>
                            Actions
                        </div>
                    </div>
                </div>

                <div className='tableBody'>
                    <div className='row'>
                        <div className='tableData'>
                        <input type="checkbox" id="check" name="selector" value=""/>
                        </div>
                        <div className='tableData'>
                           <span>2022-06-03 08:14:35</span>
                        </div>
                        <div className='tableData'>
                            <span>#885-638147-111</span>
                        </div>
                        <div className='tableData'>
                            <span> US$ 88,345.80</span>
                        </div>
                        <div className='tableData'>
                            <span>Unprocessed</span>
                        </div>
                        <div className='tableData'>
                            <button type="button" id="button5" name="edit" className={ "btnWrite " + canWrite}> NOTES</button>

                            <button type="button" id="button1" name="edit" className={ "btonEdit "+ canEdit} >EDIT</button>

                            <button type="button" id="button2" name="delete" className={"btnDelete " + canDelete}>DELETE</button>
                        </div>
                    </div>
                    <div className='row'>
                        <div className='tableData'>
                        <input type="checkbox" id="check" name="selector" value=""/>
                        </div>
                        <div className='tableData'>
                           <span>2022-05-23 04:13:33</span>
                        </div>
                        <div className='tableData'>
                            <span>#771-683447-997</span>
                        </div>
                        <div className='tableData'>
                            <span> US$ 23,341.22</span>
                        </div>
                        <div className='tableData'>
                            <span>Unprocessed</span>
                        </div>
                        <div className='tableData'>
                            <button type="button" id="button5" name="edit" className={ "btnWrite " + canWrite}> NOTES</button>

                            <button type="button" id="button1" name="edit" className={ "btonEdit "+ canEdit} >EDIT</button>

                            <button type="button" id="button2" name="delete" className={"btnDelete " + canDelete}>DELETE</button>
                        </div>
                    </div>
                    <div className='row'>
                        <div className='tableData'>
                        <input type="checkbox" id="check" name="selector" value=""/>
                        </div>
                        <div className='tableData'>
                           <span>2022-04-03 08:23:45</span>
                        </div>
                        <div className='tableData'>
                            <span>#890-632547-052</span>
                        </div>
                        <div className='tableData'>
                            <span> US$ 45,635.80</span>
                        </div>
                        <div className='tableData'>
                            <span>In Transit</span>
                        </div>
                        <div className='tableData'>
                            <button type="button" id="button5" name="edit" className={ "btnWrite " + canWrite}> NOTES</button>
                        </div>
                    </div>
                    <div className='row'>
                        <div className='tableData'>
                        <input type="checkbox" id="check" name="selector" value=""/>
                        </div>
                        <div className='tableData'>
                           <span>2022-04-03 08:23:45</span>
                        </div>
                        <div className='tableData'>
                            <span>#844-350014-244</span>
                        </div>
                        <div className='tableData'>
                            <span> US$ 121,655.80</span>
                        </div>
                        <div className='tableData'>
                            <span>In Transit</span>
                        </div>
                        <div className='tableData'>
                            <button type="button" id="button5" name="edit" className={ "btnWrite " + canWrite}> NOTES</button>
                        </div>
                    </div>


                    <div className='row'>
                        <div className='tableData'>
                        <input type="checkbox" id="check" name="selector" value=""/>
                        </div>
                        <div className='tableData'>
                           <span>2022-04-03 08:23:45</span>
                        </div>
                        <div className='tableData'>
                            <span>#222-878497-992</span>
                        </div>
                        <div className='tableData'>
                            <span> US$ 32,687.80</span>
                        </div>
                        <div className='tableData'>
                            <span>Completed</span>
                        </div>
                        <div className='tableData'>

                        </div>
                    </div>
                </div>
            </div>
        </div>

        <div className= {'container ' + canRead}>
             <span>Page 1 </span>
        </div>
    </div>
</div>)
}