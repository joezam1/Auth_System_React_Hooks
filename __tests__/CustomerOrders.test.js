import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import RouteConfig from '../configuration/routes/RouteConfig.js';

import CustomerOrders from '../src/webPages/private/customer/CustomerOrders.js';


describe('File: CustomerOrders.js', function(){
    afterEach(cleanup);
    afterEach(() => {
        jest.clearAllMocks();
    });
    describe('Component CustomerOrders', function(){
        let container;
        beforeEach(()=>{
            container = render( <MemoryRouter> <CustomerOrders/>  </MemoryRouter>  );
        });

        test('It should take a snapshot', function(){
            expect(container.asFragment(<CustomerOrders/>)).toMatchSnapshot();
        });
    });
});