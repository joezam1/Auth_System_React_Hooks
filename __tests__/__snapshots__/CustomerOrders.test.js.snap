// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: CustomerOrders.js Component CustomerOrders It should take a snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="customer-orders-section webpage hide"
  >
    <div
      class="outerLayout"
    >
      <div
        class="header-container"
      >
        <div
          class="header-title"
        >
          <span>
              Hi 
            <strong>
               
            </strong>
          </span>
        </div>
        <div
          class="header-title"
        >
          <span>
             Status: 
            <strong>
                
            </strong>
             
          </span>
        </div>
      </div>
      <div
        class="container hide"
      >
        <h2>
          Customer Orders Section.
        </h2>
        <div
          class="topNavigationBar"
        >
          <h3>
            Go To
          </h3>
          <div
            class="button-private-redirect-section hide"
          >
            <button
              class="btnRedirect"
              data-testid="button-private-redirect-id"
              type="button"
            >
                Customer Dashboard  
            </button>
          </div>
        </div>
      </div>
      <div
        class="container hide"
      >
        <button
          class="btnCreate hide"
          id="button3"
          name="add"
          type="button"
        >
           Add New Order 
        </button>
        <button
          class="btnWrite hide"
          id="button5"
          name="edit"
          type="button"
        >
           Write in Calendar
        </button>
      </div>
      <div
        class="table-container"
      >
        <div
          class="table"
        >
          <div
            class="tableHeading"
          >
            <div
              class="row"
            >
              <div
                class="tableHead"
              >
                Selector
              </div>
              <div
                class="tableHead"
              >
                Date Order
              </div>
              <div
                class="tableHead"
              >
                Order Number
              </div>
              <div
                class="tableHead"
              >
                Total Amount
              </div>
              <div
                class="tableHead"
              >
                Status
              </div>
              <div
                class="tableHead"
              >
                Actions
              </div>
            </div>
          </div>
          <div
            class="tableBody"
          >
            <div
              class="row"
            >
              <div
                class="tableData"
              >
                <input
                  id="check"
                  name="selector"
                  type="checkbox"
                  value=""
                />
              </div>
              <div
                class="tableData"
              >
                <span>
                  2022-06-03 08:14:35
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  #885-638147-111
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                   US$ 88,345.80
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  Unprocessed
                </span>
              </div>
              <div
                class="tableData"
              >
                <button
                  class="btnWrite hide"
                  id="button5"
                  name="edit"
                  type="button"
                >
                   NOTES
                </button>
                <button
                  class="btonEdit hide"
                  id="button1"
                  name="edit"
                  type="button"
                >
                  EDIT
                </button>
                <button
                  class="btnDelete hide"
                  id="button2"
                  name="delete"
                  type="button"
                >
                  DELETE
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="tableData"
              >
                <input
                  id="check"
                  name="selector"
                  type="checkbox"
                  value=""
                />
              </div>
              <div
                class="tableData"
              >
                <span>
                  2022-05-23 04:13:33
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  #771-683447-997
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                   US$ 23,341.22
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  Unprocessed
                </span>
              </div>
              <div
                class="tableData"
              >
                <button
                  class="btnWrite hide"
                  id="button5"
                  name="edit"
                  type="button"
                >
                   NOTES
                </button>
                <button
                  class="btonEdit hide"
                  id="button1"
                  name="edit"
                  type="button"
                >
                  EDIT
                </button>
                <button
                  class="btnDelete hide"
                  id="button2"
                  name="delete"
                  type="button"
                >
                  DELETE
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="tableData"
              >
                <input
                  id="check"
                  name="selector"
                  type="checkbox"
                  value=""
                />
              </div>
              <div
                class="tableData"
              >
                <span>
                  2022-04-03 08:23:45
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  #890-632547-052
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                   US$ 45,635.80
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  In Transit
                </span>
              </div>
              <div
                class="tableData"
              >
                <button
                  class="btnWrite hide"
                  id="button5"
                  name="edit"
                  type="button"
                >
                   NOTES
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="tableData"
              >
                <input
                  id="check"
                  name="selector"
                  type="checkbox"
                  value=""
                />
              </div>
              <div
                class="tableData"
              >
                <span>
                  2022-04-03 08:23:45
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  #844-350014-244
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                   US$ 121,655.80
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  In Transit
                </span>
              </div>
              <div
                class="tableData"
              >
                <button
                  class="btnWrite hide"
                  id="button5"
                  name="edit"
                  type="button"
                >
                   NOTES
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="tableData"
              >
                <input
                  id="check"
                  name="selector"
                  type="checkbox"
                  value=""
                />
              </div>
              <div
                class="tableData"
              >
                <span>
                  2022-04-03 08:23:45
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  #222-878497-992
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                   US$ 32,687.80
                </span>
              </div>
              <div
                class="tableData"
              >
                <span>
                  Completed
                </span>
              </div>
              <div
                class="tableData"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="container hide"
      >
        <span>
          Page 1 
        </span>
      </div>
    </div>
  </div>
    
</DocumentFragment>
`;
